src/main/java/com/allenai/ml/linalg/DenseVector.java

```diff
@@ -0,0 +1,64 @@
+package com.allenai.ml.linalg;
+
+import lombok.val;
+
+import java.util.Arrays;
+
+public class DenseVector implements Vector {
```

Comment:
Regarding low level numeric libraries, I'm curious what is the motivation to add ours. I'm woefully unfamiliar with numeric libraries for JVM, but I've seen things like http://nd4j.org/, which make me wonder why not adopt one of the already existing codes out there. Can you please expand?

Here is a 2008 stackoverflow link with about 7 numeric libs (if I read it correctly): http://stackoverflow.com/a/4896207/4472780 Surely there are more by now :)